### ISSUE TITLE
Z118 false positive on long attributes

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#546]
# Bug report

## What's wrong

```python
# -*- coding: utf-8 -*-

from gitlab.v4.objects import Project

def configure(project: Project) -> None:
    project.only_allow_merge_if_all_discussions_are_resolved = True
```

This code raises a false positive violation:

```
kira_setup/pipelines/project.py:27:5: Z118 Found too long name: only_allow_merge_if_all_discussions_are_resolved
```

## How is that should be

We should only check attributes for `self`, because other attributes might be a part of foreign API. That we cannot change.

**Comment by KappaDistributive** [comment]
I'd like to take on this issue.

**Comment by sobolevn** [comment]
@KappaDistributive please, feel free! Ask me any questions you have 🙂

**Comment by KappaDistributive** [comment]
@sobolevn I've obtained a first overview of the inner workings and structure of this linter.

I do have a few questions:

1. Where would it be appropriate to include this check?
My best guess right now is that it should happen in the check_name function in `/wemake_python_styleguide/visitors/ast/namping.py` but I'm not sure.

2. Is there a good resource to learn about Python `ast`?
I've looked around but the official documentation seems very incomplete, the source code hides most of the relevant part in an `_ast` that I had trouble chasing down and an internet hasn't brought up what I'm looking for thus far.

Thank you very much in advance for you help. I've been looking around for a project to contribute to regularly and I think this one might be it.

**Comment by SizovIgor** [comment]
1. I'm found TooLongNameViolation (it is Z118) in `/wemake_python_styleguide/visitors/filenames/module.py` 
class `WrongModuleNameVisitor` 
def `_check_module_name_length`

2. This is a good documentation. https://docs.python.org/3/library/ast.html

**Comment by sobolevn** [comment]
Hi, @KappaDistributive! 

Thanks for taking the time to work and think on this.
I had several conversations about this rules with my colleagues.
And we have decided that this case is valid. And nothing should be changed.

So, about your second question: we have a collection of tutorials in [`CONTRIBUTING.md`](https://github.com/wemake-services/wemake-python-styleguide/blob/master/CONTRIBUTING.md#tutorials) that's not much - but is a good start.

I really welcome new contributors, so please, take another task to work on.
Here are good and easy ones:
- https://github.com/wemake-services/wemake-python-styleguide/issues/545
- https://github.com/wemake-services/wemake-python-styleguide/issues/534

Cheers!
